lib/community/pages/create_post_page.dart

```diff
@@ -50,12 +59,31 @@ class _CreatePostPageState extends State<CreatePostPage> {
     _titleTextController.addListener(() {
       if (_titleTextController.text.isEmpty && !isSubmitButtonDisabled) setState(() => isSubmitButtonDisabled = true);
       if (_titleTextController.text.isNotEmpty && isSubmitButtonDisabled) setState(() => isSubmitButtonDisabled = false);
+
+      widget.updateDraft?.call(newDraftPost..title = _titleTextController.text);
```

Comment:
Do you think we should add a debouncer for these calls to update drafts? One downside is that it might not save the exact draft since it would be on a debouncer, but the upside is that we potentially reduce the number of calls to `updateDraft` for all cases (title, url, body, etc)

Reply:
I wasn't thinking so in this case because the callback is not actually persisting the draft, but rather just updating the in-memory representation at a higher level. Take a look at the implementation of the callback.

https://github.com/thunder-app/thunder/blob/2a51731b7eb176aa83e73537d2248847b878eb34/lib/community/widgets/community_sidebar.dart#L172

The actual persisting happens either when closing the editor without publishing or on a 10-second timer, so the impact there should be pretty minimal.

https://github.com/thunder-app/thunder/blob/2a51731b7eb176aa83e73537d2248847b878eb34/lib/community/widgets/community_sidebar.dart#L153-L157

What do you think?

Reply:
I see! That makes sense then - I was mostly worried about performance impacts on updating the shared/local preferences too many times. Just updating the memory should be okay though!